test/e2e/ingress.go

```diff
@@ -1027,3 +1027,129 @@ var _______ = describe("Ingress tests simple NLB", func() {
+		framework.ExpectNoError(err)
+
+		ing := createIngress(serviceName, hostName, ns, "/", netv1.PathTypeImplementationSpecific, labels, nil, port)
+		ingressCreate, err := cs.NetworkingV1().Ingresses(ns).Create(context.TODO(), ing, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		addr, err := jig.WaitForIngressAddress(context.TODO(), cs, ns, ingressCreate.Name, waitTime)
+		framework.ExpectNoError(err)
+
+		_, err = cs.NetworkingV1().Ingresses(ns).Get(context.TODO(), ing.Name, metav1.GetOptions{ResourceVersion: "0"})
+		framework.ExpectNoError(err)
+
+		// skipper http -> https redirect
+		By("Waiting for skipper route to default redirect from http to https, to see that our ingress-controller and skipper works")
+		err = waitForResponse(addr, "http", waitTime, isRedirect, true)
+		framework.ExpectNoError(err)
+
+		// LB ready
+		By("Waiting for ALB to create endpoint " + addr + " and skipper route, to see that our ingress-controller and skipper works")
+		err = waitForResponse(addr, "https", waitTime, isNotFound, true)
+		framework.ExpectNoError(err)
+
+		// DNS ready
+		By("Waiting for DNS to see that external-dns and skipper route to service and pod works")
+		err = waitForResponse(hostName, "https", waitTime, isSuccess, false)
+		framework.ExpectNoError(err)
+
+		// Test that we get content from the default ingress
+		By("By checking the content of the reply we see that the ingress stack works")
+		rt, quit := createHTTPRoundTripper()
+		defer func() {
+			quit <- struct{}{}
+		}()
+		url := "https://" + hostName + "/"
+		req, err := http.NewRequest("GET", url, nil)
+		framework.ExpectNoError(err)
+		resp, err := rt.RoundTrip(req)
+		framework.ExpectNoError(err)
+		s, err := getBody(resp)
+		framework.ExpectNoError(err)
+		Expect(s).To(Equal(backendContent))
+
+		// Start actual ingress tests
+		// Test ingress Filters: opaAuthorizeRequest
+
+		/**
+		## The Rule looks like below.
+		## Reference https://github.bus.zalan.do/corporate-iam/styra-smoketest-policies/blob/main/bundle/policy/ingress/rules.rego
+		default allow := false
+
+		allow if {
+		  input.attributes.request.http.method == "GET"
+		  auth_header_val := input.attributes.request.http.headers.authorization
+		  startswith(auth_header_val, "Basic ")
+		  token := substring(auth_header_val, count("Basic "), -1)
+		  token == "valid_token"
+		}
+		*/
+		path := "/"
+		opaPolicyName := "styra-smoketest"
+		updatedIng := updateIngress(ingressCreate.ObjectMeta.Name,
+			ingressCreate.ObjectMeta.Namespace,
+			hostName,
+			serviceName,
+			path,
+			netv1.PathTypeImplementationSpecific,
+			ingressCreate.ObjectMeta.Labels,
+			map[string]string{
+				"zalando.org/skipper-filter": fmt.Sprintf(`opaAuthorizeRequest("%s")`, opaPolicyName),
+			},
+			port,
+		)
+		ingressUpdate, err := cs.NetworkingV1().Ingresses(ingressCreate.ObjectMeta.Namespace).Update(context.TODO(), updatedIng, metav1.UpdateOptions{})
```

Comment:
It seems like for the test you only need an ingress. So you could drop the whole setup of deployment and service which is copied from the other tests. What @demonCoder95 was also commenting on a bit above.